instance/models/mixins/database.py

```diff
@@ -122,18 +125,47 @@ def _drop_user(cursor, user):
         cursor.execute('DROP USER %s', (user,))
 
 
+def select_random_mysql_server():
+    """
+    Helper for the field default of `mysql_server`.
+    """
+    return MySQLServer.objects.select_random()
+
+
+def select_random_mongodb_server():
+    """
+    Helper for the field default of `mongodb_server`.
+    """
+    return MongoDBServer.objects.select_random()
+
+
 # Classes #####################################################################
 
 class MySQLInstanceMixin(models.Model):
     """
     An instance that uses mysql databases
     """
     mysql_server = models.ForeignKey(
-        MySQLServer, null=True, blank=True, on_delete=models.PROTECT
+        MySQLServer,
+        null=True,
+        blank=True,
+        default=select_random_mysql_server,
+        on_delete=models.PROTECT,
     )
 
-    mysql_user = models.CharField(max_length=16, blank=True)  # 16 chars is mysql maximum
-    mysql_pass = models.CharField(max_length=32, blank=True)
+    mysql_user = models.CharField(
+        max_length=16,  # 16 chars is mysql maximum
+        # Note that the maximum length for the name of a MySQL user is 16 characters.
+        # But since we add suffixes to mysql_user to generate unique user names
+        # for different services (e.g. xqueue) we don't want to use the maximum length here.
+        default=functools.partial(get_random_string, length=6, allowed_chars=string.ascii_lowercase),
```

Comment:
Is there a reason that a partial works better or is preferred to 
```python
default=lambda: get_random_string(length=6, allowed_chars=string.ascii_lowercase)
````
? I personally find the latter more readable.